edk2toollib/uefi/edk2/parsers/base_parser_test.py

```diff
@@ -43,32 +40,80 @@ def test_process_conditional_single_boolean(self):
         # check that nothing is on the stack
         self.assertEqual(len(parser.ConditionalStack), 0)
 
+    def test_process_garbage_input(self):
+        parser = BaseParser("")
+        # make sure we fail the garbage input
+        conditional_count = len(parser.ConditionalStack)
+        self.assertFalse(parser.ProcessConditional("GARBAGE INPUT"))
+        # make sure our coutn didn't change
+        self.assertEqual(len(parser.ConditionalStack), conditional_count)
+
     def test_process_conditional_ands_ors(self):
         parser = BaseParser("")
+        self.assertTrue(parser.ProcessConditional("!if TRUE == FALSE OR TRUE == TRUE"))
+        # enable this once we have working and and or but for now just test it
+        # self.assertTrue(parser.InActiveCode())
+        # TODO: check for and and if once we've implemented this
+        # check for nested things etc
 
+    def test_process_extra_tokens(self):
+        parser = BaseParser("")
         # make sure we can't do 5 tokens
         with self.assertRaises(RuntimeError):
             parser.ProcessConditional("!if 3 == 6 ==")
 
-        # TODO: check for and and if once we've implemented this
-
-    def test_process_conditional_longer_tokens(self):
-        parser = BaseParser("")
-
         # make sure we can't do three tokens
         with self.assertRaises(RuntimeError):
             parser.ProcessConditional("!if 3 ==")
 
-        # check 4 tokens
+    def test_process_conditional_hex_number(self):
+        parser = BaseParser("")
+        # check that a hex number doesn't exqual itself
+        self.assertTrue(parser.ProcessConditional("!IF 0x30 == 30"))
+        self.assertFalse(parser.InActiveCode())
+        self.assertTrue(parser.ProcessConditional("!endif"))
+        self.assertTrue(parser.ProcessConditional("!IF 0x20 == 0x30"))
```

Comment:
should there be a passing case like 0x20 == 32?

Reply:
Looks like that doesn't work. I think we might have a bug :p